gtn/hd_span.h

```diff
@@ -125,6 +143,21 @@ class HDSpan {
     return *this;
   };
 
+  HDSpan& operator=(HDSpan&& other) {
+    if (this->data_ == other.data_) {
```

Comment:
nit: I think better to have "==" operator defined for HDSpan where we check of `data_`, 'device_`, and `size_` are the same. We can then do 
```
if (this == other)
```

Reply:
The == operator is currently defined to check for semantic equality (like the elements in the two spans are the same). This is similar to std::vector.